app/xds_api/serializers.py

```diff
@@ -124,3 +125,48 @@ def validate(self, data):
 
         # returns when user is inactive or not in the system
         raise serializers.ValidationError('Incorrect Credentials')
+
+
+class CourseSerializer(serializers.ModelSerializer):
+    """Serializes the Course model"""
+    class Meta:
+        model = Course
+        fields = ['metadata_key_hash']
+
+
+class InterestListSerializer(serializers.ModelSerializer):
+    """Serializes the interest list model"""
+    owner = XDSUserSerializer(read_only=True)
+    subscribers = XDSUserSerializer(many=True, read_only=True)
+
+    class Meta:
+        model = InterestList
+        fields = '__all__'
+
+    def create(self, validated_data):
+        name = validated_data.get("name")
+        description = validated_data.get("description")
+        owner = validated_data.get("owner")
+        return InterestList.objects.create(name=name,
+                                           description=description,
+                                           owner=owner)
+
+    def update(self, instance, validated_data):
+        instance.owner = validated_data.get('owner', instance.owner)
+        instance.description = validated_data.get('description',
+                                                  instance.description)
+        instance.name = validated_data.get('name', instance.name)
+        courses = validated_data.get('courses')
+
+        # add all new course
+        for course in courses:
+            instance.courses.add(course)
+
+        # remove any deleted course
+        for course in instance.courses.all():
+            if (course not in courses):
+                instance.courses.remove(course)
+
+        instance.save()
+
+        return instance
```

Comment:
Any exception scenarios/conditions to consider for this function?

Reply:
The exceptions thrown in the serializers are all handles in the views where they're used so we should be good on this end.